src/vue/pages/settings/Verification.vue

```diff
@@ -0,0 +1,206 @@
+<template>
+  <div class="verification">
+    <div
+      class="request-message"
+      :class="`request-message--${kycState}`"
+    >
+      <p class="request-message__content">
+        {{
+          kycRequestMessageId | globalize({
+            reason: kycRejectReason
+          })
+        }}
+      </p>
+    </div>
+    <div class="account-type">
+      <p class="account-type_label verification__label">
+        {{ 'verification-page.account-type-lbl' | globalize }}
+      </p>
+      <div class="account-type__block">
+        <router-link
+          :to="{ name: 'app.verification.general' }"
+          class="account-type__item"
+        >
+          <p class="account-type__item-title">
+            {{ 'verification-page.account-type-general-title' | globalize }}
+          </p>
+          <p class="account-type__item-description">
+            {{ 'verification-page.account-type-general-description'
+              | globalize }}
+          </p>
+          <div class="account-type__selected-icon">
+            <i class="mdi mdi-check" />
+          </div>
+        </router-link>
+        <router-link
+          :to="{ name: 'app.verification.corporate' }"
+          class="account-type__item"
+        >
+          <p class="account-type__item-title">
+            {{ 'verification-page.account-type-corporate-title' | globalize }}
+          </p>
+          <p class="account-type__item-description">
+            {{ 'verification-page.account-type-corporate-description'
+              | globalize }}
+          </p>
+          <div class="account-type__selected-icon">
+            <i class="mdi mdi-check" />
+          </div>
+        </router-link>
+      </div>
+    </div>
+    <div
+      v-if="$route.name !== 'app.verification'"
+      class="verification__form"
+    >
+      <p class="verification__label verification__form-label">
+        {{ 'verification-page.account-information-lbl' | globalize }}
+      </p>
+      <router-view />
+    </div>
+  </div>
+</template>
+
+<script>
+import { vuexTypes } from '@/vuex'
+import { mapGetters } from 'vuex'
+import { vueRoutes } from '@/vue-router/routes'
+
+import { REQUEST_STATES_STR } from '@/js/const/request-states.const'
+
+export default {
+  name: 'verification',
+  computed: {
+    ...mapGetters({
+      kycState: vuexTypes.kycState,
+      kycLatestData: vuexTypes.kycLatestData,
+      kycRejectReason: vuexTypes.kycRequestRejectReason,
+    }),
+    kycRequestMessageId () {
+      switch (this.kycState) {
+        case REQUEST_STATES_STR.pending:
+          return 'verification-page.pending-request-msg'
+        case REQUEST_STATES_STR.approved:
+          return 'verification-page.approved-request-msg'
+        case REQUEST_STATES_STR.rejected:
+          return 'verification-page.rejected-request-msg'
+        default:
+          return ''
+      }
+    },
+  },
+  beforeRouteUpdate (to, from, next) {
+    if (this.kycState && to.name === vueRoutes.verification.name) {
```

Comment:
I thought the guard in vue-routes handles it for us?

Reply:
As @quinn71 commented below, it’s better to keep these hooks in the component. Let’s do the thing with `beforeRouteEnter` and `beforeRouteUpdate`. But about using them in mixins: we have to write either comments or make the hooks assignment more obvious, please read @quinn71’s related comments below I completely agree with them.

Reply:
Resolved, updated